test/user/01-input.jl

```diff
@@ -144,18 +144,18 @@ end
 
     fw = Foodweb(:niche; S = 5, L = 5)
     @test fw.A == sparse([
+        0 0 0 1 1
+        1 1 0 0 0
+        0 0 0 0 1
         0 0 0 0 0
         0 0 0 0 0
-        0 0 0 0 0
-        1 0 0 0 0
-        0 1 1 1 1
```

Comment:
If it stay closer to the original implementation, it could be. However, I decided not to do so, first for clarity of the code. Secondly because there was an inconstancy between the niche and cascade model. For both models, we expect most of the interactions to lie in the upper triangular part which comes from the ordering of species indexes : species 1 tend to be the "top" predator that can feed on everyone below (2, 3, ... S), and so on. However, in the EcologicalNetworks implementation of the niche model, most of the ones lie on the lower triangular part (see the commented code above for instance), which is unusual. Formally it changes nothing, the structure remains the same, because you can just reverse species indices (1 becomes now the basal species and S the "top" predator).

Reply:
However, I am confident (98% let's say 😂) that my code behave as expected because I stay relatively close to the original implementation, and that modifications that I've made are minor and intended to clarify the code (and help me understand the code written in EcologicalNetworks). But, I can ofc perform more checks, even though I believe that designing such checks on random models are not so trivial.

Reply:
Indeed. Considering that these are random matrices generators, I guess the only way to test them is to draw a whole population of matrices and check their expected statistical properties. Do we have any expected statistical property?

Reply:
In any case, I think it's a good idea to at least introduce one test for one explicit `niche` random matrix and one explicit `cascade` random matrix, drawn from an explicit seed, in `./test/internals/input/test-foodwebs.jl "from structural models"`. Just to catch future breaking changes. The change introduced here should have been detected within this file in the first place.

Reply:
What we can do, is to check that on average statistical properties are similar to food webs generated by the EcologicalNetworks implementation ? 
I think of statistics like: number of producers, trophic levels, cannibalism (?), omnivory (?). What do you think about that?

Reply:
I.. think I'll trust you because this is all about the science behind niche/cascade ^ ^" But I bet this is not an easy task. If you are willing to do this, here is one neat way in 3 commits.
```
* (new) Drop `EcologicalNetworks.jl` + consistence tests.
* (new) Introduce custom `niche_model` + tests consistence with `EcologicalNetworks.nichemodel` (same for `cascade`)
* (old) Depends on EcologicalNetworks.jl